src/python/gem5/simulate/exit_event_generators.py

```diff
@@ -103,3 +104,33 @@ def default_workend_generator():
     while True:
         m5.stats.dump()
         yield False
+
+
+def default_simpoint_generator():
+    """
+    A default generator for SimPoints. It will do nothing.
+    """
+    defaultBehaviorWarning(
+        "default_simpoint_generator",
+        "A default generator for SimPoints. It will do nothing.")
+    while True:
+        yield False
+
+def simpoint_dump_reset_generator():
+    while True:
+        """
+            A generator for SimPoints. It will reset the simulation statistics
+            and then dump simulation statistics.
+        """
+        m5.stats.dump()
+        m5.stats.reset()
+        yield False
+
+def simpoint_save_checkpoint_generator(checkpoint_dir: Path):
+    while True:
+        """
+            A generator for SimPoints. It will take a checkpoint with the input
+            path and the current simulation Ticks.
+        """
```

Comment:
```suggestion
        """
            A generator for SimPoints. It will take a checkpoint with the input
            path and the current simulation Ticks.
        """
    while True:
```